assets/app/scripts/controllers/image.js

```diff
@@ -56,6 +60,51 @@ angular.module('openshiftConsole')
       );
     });
 
+    function updateTagsByName(){
+      $scope.tagsByName = {};
+      angular.forEach($scope.imageStream.spec.tags, function(tag){
+        $scope.tagsByName[tag.name] = $scope.tagsByName[tag.name] || {};
+        $scope.tagsByName[tag.name].name = tag.name;
+        $scope.tagsByName[tag.name].spec = angular.copy(tag);
+
+        // Split the name into useful parts, recalculate name to a common form to
+        // include self-references within the same IS
+        var from = $scope.tagsByName[tag.name].spec.from;
+        if (from) {
+          var splitChar = "";
+          if (from.kind === "ImageStreamImage") {
+            splitChar = "@";
+          }
+          else if (from.kind === "ImageStreamTag") {
+            splitChar = ":";
+          }
+          from._nameConnector = splitChar || null;
+          var parts = from.name.split(splitChar);
+          if (parts.length === 1) {
+            from._imageStreamName = $scope.imageStream.metadata.name;
+            from._idOrTag = parts[0];
+            from._completeName = from._imageStreamName + splitChar + from._idOrTag;
+          }
+          else if (parts.length > 1) {
+            from._imageStreamName = parts[0]; 
+            from._idOrTag = parts[1];
+            from._completeName = from._imageStreamName + splitChar + from._idOrTag;
+          }
+          else {
+            // In theory we should never get here, future proofing
```

Comment:
Actually i was being dumb, that block is impossible because either parts is going to be length 1 if it doesnt find the char, or its going to be longer than 1 ... removing that whole else block